template.yaml

```diff
@@ -44,50 +44,119 @@ Resources:
+                  ADA5MQswCQYDVQQGEwJVUzEPMA0GA1UEChMGQW1hem9uMRkwFwYDVQQDExBBbWF6
+                  b24gUm9vdCBDQSAxMB4XDTE1MDUyNjAwMDAwMFoXDTM4MDExNzAwMDAwMFowOTEL
+                  MAkGA1UEBhMCVVMxDzANBgNVBAoTBkFtYXpvbjEZMBcGA1UEAxMQQW1hem9uIFJv
+                  b3QgQ0EgMTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALJ4gHHKeNXj
+                  ca9HgFB0fW7Y14h29Jlo91ghYPl0hAEvrAIthtOgQ3pOsqTQNroBvo3bSMgHFzZM
+                  9O6II8c+6zf1tRn4SWiw3te5djgdYZ6k/oI2peVKVuRF4fn9tBb6dNqcmzU5L/qw
+                  IFAGbHrQgLKm+a/sRxmPUDgH3KKHOVj4utWp+UhnMJbulHheb4mjUcAwhmahRWa6
+                  VOujw5H5SNz/0egwLX0tdHA114gk957EWW67c4cX8jJGKLhD+rcdqsq08p8kDi1L
+                  93FcXmn/6pUCyziKrlA4b9v7LWIbxcceVOF34GfID5yHI9Y/QCB/IIDEgEw+OyQm
+                  jgSubJrIqg0CAwEAAaNCMEAwDwYDVR0TAQH/BAUwAwEB/zAOBgNVHQ8BAf8EBAMC
+                  AYYwHQYDVR0OBBYEFIQYzIU07LwMlJQuCFmcx7IQTgoIMA0GCSqGSIb3DQEBCwUA
+                  A4IBAQCY8jdaQZChGsV2USggNiMOruYou6r4lK5IpDB/G/wkjUu0yKGX9rbxenDI
+                  U5PMCCjjmCXPI6T53iHTfIUJrU6adTrCC2qJeHZERxhlbI1Bjjt/msv0tadQ1wUs
+                  N+gDS63pYaACbvXy8MWy7Vu33PqUXHeeE6V/Uq2V8viTO96LXFvKWlJbYK8U90vv
+                  o/ufQJVtMVT8QtPHRh8jrdkPSHCa2XV4cdFyQzR1bldZwgJcJmApzyMZFo6IQ6XU
+                  5MsI+yMRQ+hDKXJioaldXgjUkK642M4UwtBV8ob2xJNDd2ZhwLnoQdeXeGADbkpy
+                  rqXRfboQnoZsG4q5WTP468SQvvG5
+                  -----END CERTIFICATE-----
+                - !Ref AWS::NoValue
       Cpu: '256'
       ExecutionRoleArn: !GetAtt ghostAPIExecutionRole.Arn
       Memory: '512'
       NetworkMode: awsvpc
       RequiresCompatibilities:
         - FARGATE
-      TaskRoleArn: !If [GhostAuroraDB, !GetAtt ghostAPITaskRoleAurora.Arn, !GetAtt ghostAPITaskRoleEFS.Arn]
-      Volumes: 
+      TaskRoleArn: !If
+        - GhostAuroraDB
+        - !GetAtt ghostAPITaskRoleAurora.Arn
+        - !GetAtt ghostAPITaskRoleEFS.Arn
+      Volumes:
         - Name: ghostVolume
           EFSVolumeConfiguration:
             FilesystemId: !GetAtt ghostFS.FileSystemId
             AuthorizationConfig:
-              IAM: "ENABLED"
-            TransitEncryption: 'ENABLED'
-    DependsOn: ghostAPILogGroup
+              IAM: ENABLED
+            TransitEncryption: ENABLED
+  CloudMap:
+    Type: AWS::ServiceDiscovery::Service
+    Properties:
+      DnsConfig:
+        RoutingPolicy: MULTIVALUE
+        DnsRecords:
+          - TTL: 60
+            Type: A
+          - TTL: 60
+            Type: SRV
+      Name: 'ghost-cms'
+      NamespaceId: !Ref CloudMapNamespace
+  CloudMapNamespace:
+    Type: AWS::ServiceDiscovery::PrivateDnsNamespace
+    Properties: 
+      Name: !Sub ${AWS::StackName}-${AWS::Region}-jamstackery.io
+      Vpc: !Ref DefaultVPCId  
   Service:
     Type: AWS::ECS::Service
     Properties:
-      ServiceName: !Sub ghostAPIservice-${EnvironmentTagName}
+      ServiceName: !Sub ghostAPIservice-${AWS::StackName}
```

Comment:
We've started dropping names for most resources (that aren't S3 buckets). Might be worth just dropping it here as well (and for the IAM roles). CFN will make sure they have unique names when they are deployed.

S3 buckets are a special case due to a combination of a global namespace across all AWS accounts and the potential need to circumvent circular dependencies when there is a need to subscribe a Lambda function to Bucket events. :(

Reply:
Oh, right on! I wasn't aware they did sensibly unique naming for services automatically. Just validated.